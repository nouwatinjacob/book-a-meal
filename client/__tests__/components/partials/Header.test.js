import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  Header,
  mapStateToProps
} from '../../../src/components/partials/Header.jsx';

const props = {
  logoutAction: () => Promise.resolve(),
  isLoggedIn: true,
  isSignedup: true
};

describe('Header component -loggedin is true', () => {
  localStorage.setItem('token', customerToken);
  const setup = () => shallow(<Header {...props} />);

  it('should render Header component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handleLogout'
  when logout button is clicked`, (done) => {
    const wrapper = setup();

    const handleLogoutSpy = jest.spyOn(
      wrapper.instance(),
      'handleLogout'
    );
    wrapper.instance().handleLogout();
    expect(handleLogoutSpy).toHaveBeenCalled();

    done();
  });
});

describe('Header component -loggedin is false', () => {
  const propss = {
    logoutAction: () => Promise.resolve(),
    isLoggedIn: false,
    isSignedup: false
  };
  localStorage.setItem('token', customerToken);
  const setup = () => shallow(<Header {...propss} />);

  it(`should render Header component
   correctly when loggedin is false`, (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });
});