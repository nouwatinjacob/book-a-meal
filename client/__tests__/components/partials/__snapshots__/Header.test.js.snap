// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component -loggedin is false should render Header component
   correctly when loggedin is false 1`] = `
<div
  className="header"
>
  <div
    className="wrapper"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="Logo"
        src="https://res.cloudinary.com/sansaristic/image/upload/v1533140463/BookMeal/Default%20images/bookameal-logo.png"
      />
    </Link>
    <div
      className="header-right"
      id="myTopnav"
    >
      <Link
        className="logo"
        replace={false}
        to="/login"
      >
        SIGN IN
      </Link>
      <button
        className="button"
        id="header_button"
      >
        <Link
          replace={false}
          to="/signup"
        >
          CREATE ACCOUNT
        </Link>
      </button>
    </div>
  </div>
</div>
`;

exports[`Header component -loggedin is true should render Header component correctly 1`] = `
<div
  className="header"
>
  <div
    className="wrapper"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="Logo"
        src="https://res.cloudinary.com/sansaristic/image/upload/v1533140463/BookMeal/Default%20images/bookameal-logo.png"
      />
    </Link>
    <div
      className="header-right"
      id="myTopnav"
    >
      <Link
        className="logo"
        replace={false}
        to=""
      >
        WELCOME Â 
      </Link>
      <Link
        className="button logout"
        onClick={[Function]}
        replace={false}
        to="#"
      >
        logout
      </Link>
    </div>
  </div>
</div>
`;
